libblkid/src/superblocks/udf.c

```diff
@@ -250,18 +250,13 @@ static int probe_udf(blkid_probe pr,
 	/* The block size of a UDF filesystem is that of the underlying
 	 * storage; we check later on for the special case of image files,
 	 * which may have any block size valid for UDF filesystem */
-	uint32_t pbs[] = { 0, 512, 1024, 2048, 4096 };
-	pbs[0] = blkid_probe_get_sectorsize(pr);
+	uint32_t pbs[] = { 2048, 4096, 512, 1024, 8192, 16384, 32768 };
 
 	for (i = 0; i < ARRAY_SIZE(pbs); i++) {
 		/* Do not try with block size same as sector size two times */
 		if (i != 0 && pbs[0] == pbs[i])
```

Comment:
This is superfluous now.

Reply:
I would disagree. My program keeps failing because blkid doesn't know how to handle certain to spec block devices, which this commit helps with my supporting the only legitimately used block sizes.

Reply:
This comment referred to this specific line.
This test will never be true with your changes.

Reply:
Ah. So it is.